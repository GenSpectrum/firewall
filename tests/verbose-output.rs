use std::process::Command;

use anyhow::{anyhow, Context, Result};

#[test]
fn test_executable() -> Result<()> {
    let example_name = "main";

    let output = Command::new("cargo")
        .args([
            "run",
            "--quiet",
            "--bin",
            example_name,
            "--",
            "start",
            "--dry-run",
        ])
        .output()
        .with_context(|| anyhow!("running {example_name:?}"))?;

    assert!(output.status.success());

    assert!(output.stdout.len() == 0);

    let stderr = String::from_utf8_lossy(&output.stderr);
    let expected_stderr = include_str!("verbose-output.out");
    assert_eq!(stderr, expected_stderr);

    Ok(())
}
