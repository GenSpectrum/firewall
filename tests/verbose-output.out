+ ip6tables -t filter -D our-chain -i eth0 -j REJECT
+ ip6tables -t filter -D our-chain -i eth0 -p tcp --dport 9080 -j RETURN
+ ip6tables -t filter -D our-chain -i eth0 -p tcp --dport 80 -j RETURN
+ ip6tables -t filter -D our-chain -i eth0 -p tcp --dport 22 -j RETURN
+ ip6tables -t filter -D FORWARD -j our-chain
+ ip6tables -t filter -D INPUT -j our-chain
+ ip6tables -t filter -F our-chain
+ ip6tables -t filter -X our-chain
+ ip6tables -t filter -N our-chain
+ ip6tables -t filter -I INPUT 1 -j our-chain
+ ip6tables -t filter -I FORWARD 1 -j our-chain
+ ip6tables -t filter -A our-chain -i eth0 -p tcp --dport 22 -j RETURN
+ ip6tables -t filter -A our-chain -i eth0 -p tcp --dport 80 -j RETURN
+ ip6tables -t filter -A our-chain -i eth0 -p tcp --dport 9080 -j RETURN
+ ip6tables -t filter -A our-chain -i eth0 -j REJECT
